### PR TITLE
Adding tests to check that documentation and rules-config is correct

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
1. Adding new tests that check configurations and docs
2. Changing docs and rules-config

### CODE REVIEW

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/RulesConfigJsonTest.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ * Special test that checks that developer has not forgotten to add his warning to a rules-config.json
+ * This file is needed to be in tact with latest changes in Warnings.kt
+ */
+class RulesConfigJsonTest {
+    @Test
+    fun `read rules config json`() {
+        val allRulesFromConfig = readAllRulesFromConfig()
+        val allRulesFromCode = readAllRulesFromCode()
+
+        allRulesFromCode.forEach { rule ->
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/RulesConfigJsonTest.kt:20]: I believe we also need the opposite check - that no old warning IDs are left in config file.

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/AvailableRulesDocTest.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
+package org.cqfn.diktat.ruleset.utils
+
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.junit.Test
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/AvailableRulesDocTest.kt:4]:
```suggestion
import org.junit.jupiter.api.Test
```
Please change annotations to junit 5.

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/RulesConfigJsonTest.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package org.cqfn.diktat.ruleset.utils
+
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.common.config.rules.RulesConfigReader
+import org.cqfn.diktat.common.config.rules.getRuleConfig
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.junit.Test
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/RulesConfigJsonTest.kt:8]:
```suggestion
import org.junit.jupiter.api.Test
```
Please change annotations to junit 5.

**Review by petertrr — CHANGES_REQUESTED** [review]

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/RulesConfigJsonTest.kt (hunk 1 above)
Comment (akuleshov7) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/RulesConfigJsonTest.kt:20]: agree - will add

Comment (akuleshov7) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/RulesConfigJsonTest.kt:20]: Found 2 bugs in both config and docs, fixed!

**Review by petertrr — APPROVED** [review]
lgtm

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/AvailableRulesDocTest.kt [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * Special test to check that developer has not forgotten to write documentation for each warning.
+ * Documentation should be added to available-rules.md
+ */
+class AvailableRulesDocTest {
+    companion object {
+        const val availableRulesFile = "../info/available-rules.md"
+        const val tableDelimiter = "-----"
+        const val ruleNameHeader = "Rule name"
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/AvailableRulesDocTest.kt:15]: Maybe, it's also worth checking that rule description is not empty?

Comment (akuleshov7) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/AvailableRulesDocTest.kt:15]: will add it later

**Pull request merged**